### PR TITLE
HL-697 | Index all applications with tabs, add app to a batch

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

* Rewrite handler front page to use tabs
* Add two new views to list handled applications, both accepted and rejected ones
* Add feature to push applications to a batch (or create one if not existing)

### Front end

1. Create some applications: 
    * `docker exec -it --user appuser benefit-backend python manage.py seed --number=10`
2. Go to https://localhost:3100/ and log in
    * Marvel and look at new tabs at the front page
3. Myönnetyt / Kielletyt tabs allow you to push apps to a batch
4. Confirm that they are in a accepted / rejected batch by visiting https://localhost:8000/v1/applicationbatches/
    * Applications get bound to the first created batch with matching `proposal_for_decision` and `status="draft"`
    * If there is no existing batch, create one

### Backend

1. Run `docker exec -it --user appuser benefit-backend pytest applications/tests/test_application_batch_api.py`

### Preview

batches

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr1974.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr1974.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Review by mjturt — CHANGES_REQUESTED** [review]

File: backend/benefit/applications/api/v1/application_batch_views.py [hunk 1]
```diff
@@ -139,3 +141,71 @@ def talpa_export_batch(self, request, *args, **kwargs) -> StreamingHttpResponse:
...
+                batch = ApplicationBatch.objects.create(
+                    proposal_for_decision=matching_status
+                )
+                return batch
+
+        app_status = request.data["status"]
+        app_ids = request.data["application_ids"]
+
+        if (
+            app_status not in [ApplicationStatus.ACCEPTED, ApplicationStatus.REJECTED]
```
[the remark this answers is not in the document]
Comment (sirtawast) [reply backend/benefit/applications/api/v1/application_batch_views.py:164]: Very good! This is [implemented actually on serializer level](https://github.com/City-of-Helsinki/yjdh/blob/develop/backend/benefit/applications/api/v1/serializers/batch.py#L113) and can be removed all together as I think it's the place where this sort of logic should go in the first place.

File: backend/benefit/applications/api/v1/application_batch_views.py [hunk 2]
```diff
@@ -139,3 +141,71 @@ def talpa_export_batch(self, request, *args, **kwargs) -> StreamingHttpResponse:
         )
         approved_batches.all().update(status=ApplicationBatchStatus.SENT_TO_TALPA)
         return response
+
+    @action(methods=["POST"], detail=False)
+    @transaction.atomic
+    def create_batch(self, request):
+        def create_application_batch_by_ids(app_status, apps):
+            status_map = BatchUtils.get_matching_status_map(app_status)
+            matching_status = status_map[app_status]
```
[the remark this answers is not in the document]
Comment (sirtawast) [reply backend/benefit/applications/api/v1/application_batch_views.py:150]: Very good! This is [implemented actually on serializer level](https://github.com/City-of-Helsinki/yjdh/blob/develop/backend/benefit/applications/api/v1/serializers/batch.py#L113) and can be removed all together as I think it's the place where this sort of logic should go in the first place.

File: backend/benefit/applications/tests/test_application_batch_api.py [hunk 3]
```diff
@@ -488,6 +559,22 @@ def test_application_batch_export(mock_export, handler_api_client, application_b
     assert response.status_code == 400
 
 
+def test_matching_status_map():
```
[the remark this answers is not in the document]
Comment (sirtawast) [reply backend/benefit/applications/tests/test_application_batch_api.py:562]: Removed, good practice round 😂

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr1974.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Review by mjturt — APPROVED** [review]
:+1: 

Maybe remove the commented code in the test before merge.

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr1974.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe employer result is __failed__ for https://kesaseteli-pr1974.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr1974.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr1974.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr1974.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**